fluent/context.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import absolute_import, unicode_literals
+
+import babel
+import babel.numbers
+import babel.plural
+import six
+
+from .builtins import BUILTINS
+from .resolver import resolve
+from .syntax import FluentParser
+from .syntax.ast import Message, Term
+from .utils import cachedproperty
+
+
+class MessageContext(object):
+    """
+    Message contexts are single-language stores of translations.  They are
+    responsible for parsing translation resources in the Fluent syntax and can
+    format translation units (entities) to strings.
+
+    Always use `MessageContext.format` to retrieve translation units from
+    a context.  Translations can contain references to other entities or
+    external arguments, conditional logic in form of select expressions, traits
+    which describe their grammatical features, and can use Fluent builtins.
+    See the documentation of the Fluent syntax for more information.
+    """
+
+    def __init__(self, locales, functions=None, use_isolating=True):
+        self.locales = locales
+        _functions = BUILTINS.copy()
+        if functions:
+            _functions.update(functions)
+        self._functions = _functions
+        self._use_isolating = use_isolating
+        self._messages = {}
+        self._terms = {}
+
+    def add_messages(self, source):
+        parser = FluentParser()
+        resource = parser.parse(source)
+        # TODO - warn if items get overwritten
```

Comment:
actually, we reject items if they would override already existing element, and we report the result.

https://github.com/projectfluent/fluent-rs/blob/master/fluent/src/bundle.rs#L141

I would recommend looking at Rust signatures as they're stricter about what they can/should accept and return.

Reply:
OK, thanks, that is helpful.

In my later branch (which adds a compile-to-Python implementation), I have added better handling for this case, at least in terms of collecting the error  - 
https://github.com/projectfluent/python-fluent/pull/71/files#diff-f1879dcf9535775ee300cd035561340dR44

and it includes an API for actually seeing the errors (whether at the parsing or compiling stage). 

If I fix this to not overwrite, and leave the reporting of the error to my other branch (which already reports it), would that be OK?